source/admin_manual/list_of_events.rst

```diff
@@ -1028,6 +1043,32 @@ Mail
 | uid                 | Mail IMAP UID number                                 |
 +---------------------+------------------------------------------------------+
 
+.. _event_mail_opened:
+
+mail_opened
+^^^^^^^^^^^
+
+.. versionadded:: 2.3.16
+
+A mail is opened.
```

Comment:
Should explain exactly what "opened" means. Actually, I'm not sure what the current implementation does. Is mail_opened sent when dbox opens the file for reading metadata?

Reply:
Isn't it usually the mailbox that might get opened for metadata rather than mails? I don't see this event being emiited with IMAP get/setmetadata.

Reply:
There's many kinds of metadata :) I didn't mean the mailbox metadata, but dbox has e.g. the received-date in an internal metadata header. So if the received-date isn't in dovecot.index.cache, then FETCH INTERNALDATE in IMAP will cause the dbox file to be opened and the metadata header read.

Reply:
Right. Yeah, managed to make the event emit on reading save date and flags. I've now added a sentence to give a hint when mail files might be opened. Is it good enough?

Reply:
Yes, except .. flags? Message flags are in the index, not in the message body. They shouldn't cause this event.

Reply:
And are you sure it's emitted even for metadata? Looking at the dbox code, it seems to me it's not sent when opening file for metadata. And if not, it would still be good to specifically mention that it is not emitted when opening due to reading metadata.

Reply:
Oh, right. It's emitting an event for opening mailbox not mail.

Though IMO it's weird that mailbox_open_full() doesn't set event name. Also, if reason is NULL it should still emit the event.